packages/mermaid/src/diagrams/xychart/xychartRenderer.ts

```diff
@@ -0,0 +1,140 @@
+import { select } from 'd3';
+import { Diagram } from '../../Diagram.js';
+import * as configApi from '../../config.js';
+import { log } from '../../logger.js';
+import { configureSvgSize } from '../../setupGraphViewbox.js';
+import {
+  DrawableElem,
+  TextElem,
+  TextHorizontalPos,
+  TextVerticalPos,
+} from './chartBuilder/Interfaces.js';
+
+export const draw = (txt: string, id: string, _version: string, diagObj: Diagram) => {
+  function getDominantBaseLine(horizontalPos: TextHorizontalPos) {
+    return horizontalPos === 'top' ? 'hanging' : 'middle';
+  }
+
+  function getTextAnchor(verticalPos: TextVerticalPos) {
+    return verticalPos === 'left' ? 'start' : verticalPos === 'right' ? 'end' : 'middle';
+  }
+
+  function getTextTransformation(data: TextElem) {
+    return `translate(${data.x}, ${data.y}) rotate(${data.rotation || 0})`;
+  }
+  const conf = configApi.getConfig();
+
+  log.debug('Rendering xychart chart\n' + txt);
+
+  const securityLevel = conf.securityLevel;
+  // Handle root and Document for when rendering in sandbox mode
+  let sandboxElement;
+  if (securityLevel === 'sandbox') {
+    sandboxElement = select('#i' + id);
+  }
+  const root = sandboxElement ? sandboxElement : select('body');
+
+  const svg = root.select(`[id="${id}"]`);
+
+  const group = svg.append('g').attr('class', 'main');
+
+  const width = 700;
+  const height = 500;
+
```

Comment:
This part was recently turned into a [separate function](https://github.com/mermaid-js/mermaid/blob/43217e1395113d2f01b2899842389993955e0a6b/packages/mermaid/src/rendering-util/selectSvgElement.ts#L11C1-L11C1), please use it instead. Very likely the only thing you really need as a result of that code is  a `svg` variable solely

Reply:
Thanks! updated